documentation/adr/0023-error-format.md

```diff
@@ -0,0 +1,74 @@
+
+Date: 2021-Mar-24
+
+## Status
+
+Accepted
+
+## Context
+
+Currently throughout PACE, different error identifiers are used to reference the same error.
+This makes it difficult in cases of specific try-catch (as described in [ADR 21](./0021-errors-and-warnings.md)) to know what to compare against.
+
+Examples include:
+```
+herbert_core/applications/multifit/@mfclass/fit.m                                             : Herbert:mfclass:invalid_argument
+herbert_core/classes/data_loaders/@rundata/private/gen_runfiles_.m                            : GEN_GRUNFILES:invalid_arguments
+herbert_core/utilities/general/catstruct.m                                                    : catstruct:InvalidArgument
+herbert_core/classes/data_loaders/parse_arg.m                                                 : PARSE_ARG:wrong_arguments
+
+herbert_core/classes/data_loaders/@rundata/get_par.m                                          : RUNDATA:invalid_argument
+herbert_core/classes/data_loaders/@rundata/get_rundata.m                                      : RUNDATA:invalid_arguments
+
+herbert_core/utilities/maths/noisify.m                                                        : HERBERT:noisify
+herbert_core/utilities/misc/objdiff.m                                                         : YMA:OBJDIFF:NotEnoughInputs
+herbert_core/utilities/xml_io_tools/xmlwrite_xerces.m                                         : xml:FileNotFound
+herbert_core/admin/matlab_nbits.m                                                             : MATLAB:NBITS
+
+_test/shared/matlab_xunit_ISISextras/@TestCaseWithSave/private/get_ref_dataset_.m             : TestCaseWithSave:invalid_argument
+_test/shared/matlab_xunit_ISISextras/@TestCaseWithSave/private/instantiate_methods_to_save_.m : TEST_CASE_WITH_SAVE:invalid_argument
+
+erbert_core/classes/MPIFramework/@iMessagesFramework/iMessagesFramework.m                     : iMESSAGES_FRAMEWORK:invalid_argument
+herbert_core/classes/MPIFramework/@iMessagesFramework/iMessagesFramework.m                    : MESSAGES_FRAMEWORK:invalid_argument
+herbert_core/classes/MPIFramework/@iMessagesFramework/private/mix_messages_.m                 : iMESSAGES_FRAMEWOR:invalid_argument
+```
+From which it should be obvious is not a consistent scheme for error identification, even within the same set of routines.
+
+## Decision
+
+As MATLAB standards do not enforce a particular style of error statement,
+this document seeks to outline the format an error identifier should take as well as establish a list of standard common error identifiers.
+
+Following discussions it was decided that it would be useful to have a standard format with 3 components, in the following format
+expanding upon that intially outlined in [ADR 21](./0021-errors-and-warnings.md).
+
+
+```matlab
+error('(HORACE|HERBERT):(function_name|ClassName):error_identifier')
+```
+
+This scheme is based on the current (2020b) MATLAB error identifiers common to many functions.
```

Comment:
May be worth linking to MATLAB documentation (in the context Section) on which this template is based: https://uk.mathworks.com/help/matlab/ref/mexception.html definition of `errId`